src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -460,11 +459,11 @@ private void NotifyFileSystemEventArgs(WatcherChangeTypes changeType, ReadOnlySp
         /// </summary>
         private void NotifyFileSystemEventArgs(WatcherChangeTypes changeType, string name)
         {
-            FileSystemEventHandler? handler = GetHandler(changeType);
+            Action<FileSystemEventArgs>? raiser = GetRaiser(changeType);
 
-            if (handler != null && MatchPattern(string.IsNullOrEmpty(name) ? _directory : name))
+            if (raiser != null && MatchPattern(string.IsNullOrEmpty(name) ? _directory : name))
             {
-                handler(this, new FileSystemEventArgs(changeType, _directory, name));
+                raiser(new FileSystemEventArgs(changeType, _directory, name));
             }
         }
```

Comment:
You're correct that it's not being used right now.  However, I think that's arguably a perf bug.  Call sites that already have a string (but stored in a span) are then calling the span-based overload, and then having to call ToString on the span to produce a new one.  It'd be better if those call sites, if they have a string, would actually use the string-based overload, or potentially they'd be consolidated into a single method that accepted both a span and an optional string, using the string instead of ToString if the string is non-null.

That's not a change for this PR, but it's to say let's not remove this yet.